cloudinary-test-common/src/main/java/com/cloudinary/test/AbstractStructuredMetadataTest.java

```diff
@@ -0,0 +1,241 @@
+            try {
+                api.deleteMetadataField(externalId);
+            } catch (Exception ignored) {
+            }
+        }
+    }
+
+    @Rule
+    public TestName currentTest = new TestName();
+
+    @Before
+    public void setUp() {
+        System.out.println("Running " + this.getClass().getName() + "." + currentTest.getMethodName());
+        this.cloudinary = new Cloudinary();
+        assumeNotNull(cloudinary.config.apiSecret);
+        this.api = cloudinary.api();
+    }
+
+    @Test
+    public void testCreateMetadata() throws Exception {
+        StringMetadataField stringField = newFieldInstance("testCreateMetadata_1");
+        ApiResponse result = addFieldToAccount(stringField);
+        assertNotNull(result);
+        assertEquals(stringField.getLabel(), result.get("label"));
+
+        SetMetadataField setField = createSetField("testCreateMetadata_2");
+        result = cloudinary.api().addMetadataField(setField);
+        assertNotNull(result);
+        assertEquals(setField.getLabel(), result.get("label"));
+    }
+
+    @Test
+    public void testDateFieldBehaviour() throws Exception {
+        // now minus 3 days hours.
+        Date max = new Date();
+        Date min = new Date(max.getTime() - 72 * 60 * 60 * 1000);
+
+        Date legalValue = new Date(min.getTime() + 36 * 60 * 60 * 1000);
+        Date illegalValue = new Date(max.getTime() + 36 * 60 * 60 * 1000);
+
+        DateMetadataField dateMetadataField = new DateMetadataField();
+        dateMetadataField.setLabel("Start date" + new Date().getTime());
+
+        List<MetadataValidation> rules = new ArrayList<MetadataValidation>();
+        rules.add(new MetadataValidation.DateGreaterThan(min));
+        rules.add(new MetadataValidation.DateLessThan(max));
+        dateMetadataField.setValidation(new MetadataValidation.AndValidator(rules));
+
+        String message = null;
+        ApiResponse res = null;
+        try {
+            // should fail
+            dateMetadataField.setDefaultValue(illegalValue);
+            res = api.addMetadataField(dateMetadataField);
+            // this line should not be reached if all is working well, but when it's not we still want to clean it up:
+            metadataFieldExternalIds.add(res.get("external_id").toString());
+        } catch (BadRequest e) {
+            message = e.getMessage();
+        }
+
+        assertEquals(message, "default_value is invalid");
+
+        // should work:
+        dateMetadataField.setDefaultValue(legalValue);
+        res = api.addMetadataField(dateMetadataField);
+        metadataFieldExternalIds.add(res.get("external_id").toString());
+    }
+
+    @Test
+    public void testListFields() throws Exception {
+        StringMetadataField stringField = newFieldInstance("testListFields");
+        addFieldToAccount(stringField);
+
+        ApiResponse result = cloudinary.api().listMetadataFields();
+        assertNotNull(result);
```

Comment:
Let's assert the rest

Reply:
I can assert that there's a list, not the actual item as it may be far behind and not get fetched on the first batch (parallel testing).

Reply:
Not sure I get what you mean...